spec/spec_helper.rb

```diff
@@ -3,10 +3,67 @@
 require 'json'
 
 RSpec.configure do |config|
-  # Disable RSpec exposing methods globally on `Module` and `main`
+  # Disable RSpec exposing methods globally on `Module` and `main`.
   config.disable_monkey_patching!
 
   config.expect_with :rspec do |c|
     c.syntax = :expect
   end
 end
+
+# Forces serialized models to use UTC for date and times. This is used for comparing date
+# and times between JSON versions of records (i.e. useful for testing ONLY).
+def to_utc(contents)
+  date_time_pattern = /\d{4}-[01]\d-[0-3]\dT[0-2]\d:[0-5]\d:[0-5]\d\.\d+([+-][0-2]\d:[0-5]\d|Z)/
+  contents.gsub(date_time_pattern) do |match|
+    DateTime.parse(match).new_offset(0).to_s
+  end
+end
+
+# Helper method to strip fields we don't care about comparing.
+def ignore_irrelavant_fields(json)
+  ignore = ['created_at', 'updated_at', 'bundle_id', 'status_code']
+  # Unused Encounter fields (start_time and end_time are used instead).
+  ignore += ['admitTime', 'dischargeTime', 'transferFrom']
+  # Unnecessary Bonnie stuff (measure history, patient bank, expired).
+  ignore += ['calc_results', 'has_measure_history', 'results_exceed_storage', 'results_size', 'version', 'is_shared', 'expired']
+  # Unused Medication fields.
+  ignore += ['fulfillmentHistory', 'administrationTiming', 'allowedAdministrations']
+  ignore.each { |ignore_key| json.deep_reject_key!(ignore_key) }
+  json
+end
+
+# Extension to Hash to deep delete the given key (recusive on any nested Hashes and
+# Arrays that might be in the has that deep_reject_key! is called on).
+class Hash
+  def deep_reject_key!(key)
+    keys.each { |k| delete(k) if k == key || self[k] == self[key] }
+    values.each { |v| v.deep_reject_key!(key) if v.is_a? Hash }
+    values.each do |v|
+      next unless v.is_a? Array
+      v.each do |el|
+        el.deep_reject_key!(key) if el.is_a? Hash
+      end
+    end
+    self
+  end
+end
+
+# Extension to Hash to deep clean hashes for test comparison purposes. This makes sure
+# things are always in order, specifically for some tricky cases with data elements.
+#
+# This makes: {a: 'b', c: 'd'} == {c: 'd', a: 'b'}
+class Hash
+  def clean_hash(&block)
```

Comment:
If `clean_hash` is acting on the object it's called on, by Ruby-logic should it be named `clean_hash!`? Like `deep_reject_key!` above